src/cli/llm_client.rs

```diff
@@ -113,4 +128,62 @@ impl OpenAIClient {
 
         Ok(rec)
     }
+
+    async fn ask(&self, question: String, context: Vec<String>) -> Result<String, ShellError> {
+        let mut messages: Vec<ChatCompletionRequestMessage> = vec![];
+
+        messages.push(
+            ChatCompletionRequestSystemMessageArgs::default()
+                .content("You are a helpful assistant.")
+                .build()
+                .unwrap()
+                .into(),
+        );
+
+        for ctx in context {
+            messages.push(
+                ChatCompletionRequestSystemMessageArgs::default()
+                    .content(ctx)
+                    .build()
+                    .unwrap()
+                    .into(),
+            )
+        }
+
+        messages.push(
+            ChatCompletionRequestUserMessageArgs::default()
+                .content(question)
+                .build()
+                .unwrap()
+                .into(),
+        );
+
+        let client = Client::with_config(
+            async_openai::config::OpenAIConfig::default().with_api_key(self.api_key.clone()),
+        );
+
+        let request = CreateChatCompletionRequestArgs::default()
+            .max_tokens(512u16)
+            .model("gpt-3.5-turbo")
+            .messages(messages)
+            .build()
+            .unwrap();
+
+        let response = client.chat().create(request).await;
+
+        let answer = match response {
+            Ok(r) => r.choices[0].message.content.as_ref().unwrap().to_string(),
```

Comment:
What happens if the response has no choices? Is that possible?

Reply:
I don't think this is possible. You get a response for each message sent, and we are sending one, so will always be a choice. If there is no message, which there can't be, or another issue with the request then we end up in the Err() branch.